crates/providers/src/provider.rs

```diff
@@ -184,6 +184,13 @@ pub trait TempProvider: Send + Sync {
         keys: Vec<StorageKey>,
         block: Option<BlockId>,
     ) -> TransportResult<EIP1186AccountProofResponse>;
+    /// Inspect and debug transaction
+    async fn debug_trace_call(
+        &self,
+        req: TransactionRequest,
+        block: Option<BlockId>,
+        trace_options: GethDebugTracingCallOptions,
```

Comment:
this should be encapsulated in a helper, builder style struct, which would make calling this a bit easier

Reply:
> this should be encapsulated in a helper, builder style struct, which would make calling this a bit easier

interesting, so i should not follow how it is implemented in ethers-rs ?

 https://github.com/gakonst/ethers-rs/blob/6136ddeb7227df1ffaf792a7518d7e3a580eb573/ethers-providers/src/middleware.rs#L746

Reply:
I'm fine with this but it would be more convenient to a builder-style request object for this

Reply:
> I'm fine with this but it would be more convenient to a builder-style request object for this

based, i work on builder pattern for this then ! thanks

Reply:
> I'm fine with this but it would be more convenient to a builder-style request object for this

made changes